packages/mobile/src/send/reducers.ts

```diff
@@ -1,18 +1,50 @@
+import BigNumber from 'bignumber.js'
+import { TokenTransactionType } from 'src/apollo/types'
 import { areRecipientsEquivalent, Recipient } from 'src/recipients/recipient'
 import { getRehydratePayload, REHYDRATE, RehydrateAction } from 'src/redux/persist-helper'
+import { RootState } from 'src/redux/reducers'
 import { Actions, ActionTypes } from 'src/send/actions'
 
 // Sets the limit of recent recipients we want to store
 const RECENT_RECIPIENTS_TO_STORE = 8
 
+export interface TransactionData {
+  recipient: Recipient
+  amount: BigNumber
+  reason: string
+  type: TokenTransactionType
+  firebasePendingRequestUid?: string | null
+}
+
+export interface ConfirmationInput {
+  recipient: Recipient
+  amount: BigNumber
+  reason: string
+  recipientAddress: string | null | undefined
+  type: TokenTransactionType
+  firebasePendingRequestUid?: string | null
+}
+
+export interface SecureSendPhoneNumberMapping {
+  [e164Number: string]: {
+    address: string | undefined
+    addressValidationRequired: boolean
+    fullValidationRequired: boolean
+  }
+}
+
 export interface State {
   isSending: boolean
   recentRecipients: Recipient[]
+  isValidRecipient: boolean
+  secureSendPhoneNumberMapping: SecureSendPhoneNumberMapping
```

Comment:
I think the identity reducer is a better place for this. It's coupled more closely to contact mapping than to the send flow IMO. Keep in mind this will eventually affect other parts of the app

Reply:
Where else do you think this could come into play? See comments below about why I think using this outside of Secure Send is an issue.

Reply:
I guess it's more a matter of opinion really but the name 'Secure Send' is just marketing. It's really 'Attestation contract mapping disambiguation'. All of the other code dealing with the Attestation contract, verification, and the creation of number<->address mappings lives in the identity folder.